src/restartable.js

```diff
@@ -169,7 +169,11 @@ export default function restartable (driver, opts = {}) {
       }
     }
 
-    lastSinkEvent$.imitate(sink$);
+    if (!(sink$ instanceof MemoryStream)) {
```

Comment:
Ah yeah that's fair enough, I had it in the conditional because I wasn't sure if there'd be any performance impact or other unintended side-effects by always doing that, but it seems to work fine.

Updated in latest commit to always call `.filter()` on `sink$` when imitating it.